snakemake/shell.py

```diff
@@ -29,6 +30,14 @@
     STDOUT = None
 
 
+# There is a max length for a command executed as well as a maximum
+# length for each argument passed to a command. The latter impacts us
+# especially when doing `sh -c 'long script from user'`. On Linux, it's
+# hardcoded in the kernel as 32 pages, or 128kB. On OSX it appears to be
+# close to `getconf ARG_MAX`, about 253kb.
+MAX_ARG_LEN = 16 * 4096 - 1
```

Comment:
Using 16 * 4096 instead of 32*4096 to be conservative. A script that's 64kb long we may as well pass as a file. Personally, somewhere at 1kb I'd say please don't spam this to the log...